src/Swarm/Util/Location.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE PatternSynonyms #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- Orphan JSON instances for Location and Heading
+
+-- |
+-- Module      :  Swarm.Util.Location
+-- Copyright   :  Brent Yorgey
+-- Maintainer  :  byorgey@gmail.com
+--
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Locations and headings.
+module Swarm.Util.Location (Location, pattern Location, Heading, origin, Affine (..)) where
+
+import Data.Aeson (FromJSONKey, ToJSONKey)
+import Data.Int (Int32)
+import Data.Yaml (FromJSON (parseJSON), ToJSON (toJSON))
+import Linear (V2 (..))
+import Linear.Affine (Affine (..), Point (..), origin)
+
+-- | A Location is a pair of (x,y) coordinates, both up to 32 bits.
+--   The positive x-axis points east and the positive y-axis points
+--   north.
+type Location = Point V2 Int32
```

Comment:
Maybe add some documentation here about why `Coords` is a different type than `Location`, and precisely how the conventions differ.

Reply:
Good idea! :heavy_check_mark: